### PR TITLE
Narc indexes

### PR DESCRIPTION
**Pull request by wildfire248**
Adds an enum for referencing the NARCs in associated functions, and changes any obvious magic numbers.

## Renamed Constants

- `src/narc.c`: `Unk_02100498[]` -> `NarcPaths[]`

## New Constants

- `include/constants/narc.h` - Enum for indexing into the NARC path table.
    - (Note: enum names were just based on the paths with no changes, there are almost definitely better names for these.)

## Other Renames

- Replaced any obvious magic numbers with enum values.

### CODE REVIEW

File: include/constants/narc.h [hunk 1]
```diff
@@ -0,0 +1,207 @@
+#ifndef POKEPLATINUM_CONSTANTS_NARC_H
+#define POKEPLATINUM_CONSTANTS_NARC_H
+
+
+
```
Comment (mid-kid) [inline include/constants/narc.h:5]: One newline is enough

File: include/constants/narc.h [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+/*
+ * Indexes into the NARC path table, make sure these match the table in narc.c
+ */
```
Comment (mid-kid) [inline include/constants/narc.h:8]: This makes more sense as a C++ style comment.

File: src/narc.c [hunk 3]
```diff
@@ -4,7 +4,10 @@
 #include "narc.h"
 #include "heap.h"
 
-static const char * Unk_02100498[] = {
+/*
+ * NARC path table, make sure index enum in constants/narc.h matches
+ */
```
Comment (mid-kid) [inline src/narc.c:9]: Idem.

**Review by mid-kid — CHANGES_REQUESTED** [review]

**Comment by mid-kid** [comment]
Please squash your "Replace static NARC index numbers with enum values" commits.

**Comment by wildfire248** [comment]
Squashed the magic number replacements and fixed the comments/newlines.

**Pull request merged**